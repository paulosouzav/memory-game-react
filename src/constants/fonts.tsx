import { css } from 'styled-components';
import Roboto from '../assets/fonts/Roboto-Regular.ttf';
import RobotoThin from '../assets/fonts/Roboto-Thin.ttf';
import RobotoLight from '../assets/fonts/Roboto-Light.ttf';
import RobotoBold from '../assets/fonts/Roboto-Bold.ttf';

export const FONTS = Object.freeze({
  REGULAR: 'Roboto',
  THIN: 'RobotoThin',
  LIGHT: 'RobotoLight',
  BOLD: 'RobotoBold',
});

interface FontFace {
  name: string;
  url: any;
  weight: string | number;
}

const FONT_FACES_LIST: Array<FontFace> = [
  {
    name: FONTS.REGULAR,
    url: Roboto,
    weight: 'normal',
  },
  {
    name: FONTS.THIN,
    url: RobotoThin,
    weight: 100,
  },
  {
    name: FONTS.LIGHT,
    url: RobotoLight,
    weight: 300,
  },
  {
    name: FONTS.BOLD,
    url: RobotoBold,
    weight: 'bold',
  },
];

export const FONT_FACES = css`
  ${FONT_FACES_LIST.map(
    ({ name, url, weight }: FontFace) => css`
      @font-face {
        font-family: ${name}, 'sans-serif';
        src: url(${url});
        font-weight: ${weight};
      }
    `
  )}
`;
