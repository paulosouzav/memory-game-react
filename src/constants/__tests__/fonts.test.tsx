import { FONTS, FONT_FACES } from '../fonts';

describe('FONTS', () => {
  it('should have expected values', () => {
    expect(FONTS).toMatchSnapshot();
  });
});

describe('FONT_FACES', () => {
  it('should have expected values', () => {
    expect(FONT_FACES).toMatchSnapshot();
  });
});
