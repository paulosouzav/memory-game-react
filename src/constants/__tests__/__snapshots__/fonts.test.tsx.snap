// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FONT_FACES should have expected values 1`] = `
Array [
  "
  ",
  "
      @font-face {
        font-family: ",
  "Roboto",
  ", 'sans-serif';
        src: url(",
  "Roboto-Regular.ttf",
  ");
        font-weight: ",
  "normal",
  ";
      }
    ",
  "
      @font-face {
        font-family: ",
  "RobotoThin",
  ", 'sans-serif';
        src: url(",
  "Roboto-Thin.ttf",
  ");
        font-weight: ",
  "100",
  ";
      }
    ",
  "
      @font-face {
        font-family: ",
  "RobotoLight",
  ", 'sans-serif';
        src: url(",
  "Roboto-Light.ttf",
  ");
        font-weight: ",
  "300",
  ";
      }
    ",
  "
      @font-face {
        font-family: ",
  "RobotoBold",
  ", 'sans-serif';
        src: url(",
  "Roboto-Bold.ttf",
  ");
        font-weight: ",
  "bold",
  ";
      }
    ",
  "
",
]
`;

exports[`FONTS should have expected values 1`] = `
Object {
  "BOLD": "RobotoBold",
  "LIGHT": "RobotoLight",
  "REGULAR": "Roboto",
  "THIN": "RobotoThin",
}
`;
